cmd/migrate_from_redis.go

```diff
@@ -0,0 +1,243 @@
+	return uint64(info.NumDocs), nil
+}
+
+func (r *MigrateFromRedisCmd) migrateData(ctx context.Context, rdb *redis.Client, targetClient *qdrant.Client, sourcePointCount uint64) error {
+	batchSize := r.Migration.BatchSize
+
+	var currentOffset uint64 = 0
+
+	if !r.Migration.Restart {
+		_, offsetStored, err := commons.GetStartOffset(ctx, r.Migration.OffsetsCollection, targetClient, r.Redis.Index)
+		if err != nil {
+			return fmt.Errorf("failed to get start offset: %w", err)
+		}
+		currentOffset = offsetStored
+	}
+
+	bar, _ := pterm.DefaultProgressbar.WithTotal(int(sourcePointCount)).Start()
+	displayMigrationProgress(bar, currentOffset)
+
+	info, err := rdb.FTInfo(ctx, r.Redis.Index).Result()
+	if err != nil {
+		return fmt.Errorf("failed to get index info: %w", err)
+	}
+
+	attrTypes := make(map[string]string)
+	for _, attr := range info.Attributes {
+		attrTypes[attr.Identifier] = attr.Type
+	}
+
+	for {
+		res, err := rdb.FTSearchWithArgs(ctx, r.Redis.Index, "*", &redis.FTSearchOptions{
+			LimitOffset: int(currentOffset),
+			Limit:       int(batchSize),
+		}).Result()
+		if err != nil {
+			return fmt.Errorf("failed to search Redis: %w", err)
+		}
+
+		count := len(res.Docs)
+		if count == 0 {
+			break
+		}
+
+		targetPoints := make([]*qdrant.PointStruct, 0, count)
+
+		for i := 0; i < count; i++ {
+			doc := res.Docs[i]
+
+			parsedFields := make(map[string]interface{})
+			vectorMap := make(map[string]*qdrant.Vector)
+
+			for fieldName, rawVal := range doc.Fields {
+				attrType := attrTypes[fieldName]
+
+				if attrType == redis.SearchFieldTypeVector.String() {
+					vec := bytesToFloats([]byte(rawVal))
+					vectorMap[fieldName] = qdrant.NewVectorDense(vec)
+				} else {
+					parsedFields[fieldName] = parseFieldValue(attrType, rawVal)
+				}
+			}
+
+			point := &qdrant.PointStruct{
+				Id:      arbitraryIDToUUID(doc.ID),
+				Vectors: qdrant.NewVectorsMap(vectorMap),
+			}
+
+			payload := qdrant.NewValueMap(parsedFields)
+			payload[r.IdField] = qdrant.NewValueString(doc.ID)
+			point.Payload = payload
+
+			targetPoints = append(targetPoints, point)
+		}
+
+		if len(targetPoints) > 0 {
+			_, err = targetClient.Upsert(ctx, &qdrant.UpsertPoints{
+				CollectionName: r.Qdrant.Collection,
+				Points:         targetPoints,
+				Wait:           qdrant.PtrOf(true),
+			})
+			if err != nil {
+				return fmt.Errorf("failed to insert data into target: %w", err)
+			}
+		}
+
+		currentOffset += uint64(count)
+		// Just a placeholder ID for offset tracking.
+		// We're only using the offset count
+		offsetId := qdrant.NewIDNum(0)
+		err = commons.StoreStartOffset(ctx, r.Migration.OffsetsCollection, targetClient, r.Redis.Index, offsetId, currentOffset)
+		if err != nil {
+			return fmt.Errorf("failed to store offset: %w", err)
+		}
+
+		bar.Add(count)
+	}
+
+	pterm.Success.Printfln("Data migration finished successfully")
+	return nil
+}
+
+func bytesToFloats(b []byte) []float32 {
```

Comment:
Inverse of the helper documented at https://redis.io/docs/latest/develop/clients/go/vecsearch/#:~:text=Add%20a%20helper%20function.